src/idl_parser.cpp

```diff
@@ -2737,17 +2739,18 @@ CheckedError Parser::ParseProtoDecl() {
   return NoError();
 }
 
-CheckedError Parser::StartEnum(const std::string &enum_name, bool is_union,
+CheckedError Parser::StartEnum(const std::string &name, bool is_union,
                                EnumDef **dest) {
   auto &enum_def = *new EnumDef();
-  enum_def.name = enum_name;
+  enum_def.name = name;
   enum_def.file = file_being_parsed_;
   enum_def.doc_comment = doc_comment_;
   enum_def.is_union = is_union;
   enum_def.defined_namespace = current_namespace_;
-  if (enums_.Add(current_namespace_->GetFullyQualifiedName(enum_name),
-                 &enum_def))
-    return Error("enum already exists: " + enum_name);
+  if (enums_.Add(current_namespace_->GetFullyQualifiedName(name), &enum_def))
+    return Error("enum already exists: " + name);
+  if (structs_.Lookup(name) != nullptr)
```

Comment:
Probably this lookup will check only predeclared structs or structs in the global namespace while enum will be added into `current_namespace_`.
The same for the enum lookup in the StartStruct.
It would be better to add tests for clashes in different namespaces and clashes with predeclared structs as well.

Reply:
Sure, I'll look into writing some tests with that in mind. Thanks for the feedback.